### DISCUSSION TITLE
Regions plugin: new random color on each new region with drag selection enabled

### DISCUSSION THREAD

**Discussion by thomasmol** [katspaugh/wavesurfer.js#3261] [Q&A] [answered]
Currently the color in the Regions plugin in the enableDragSelection method cannot be 'dynamic'. E.g. -> in the example (https://wavesurfer-js.org/examples/?regions.js) there is a randomColor() method used to generate a random color, however each new region created by dragging will use the same color (generated by the initial invoke of the randomColor() method). Right now I am currently doing this to have a random color for each new region, but seems a bit cumbersome: 

```typescript
wsRegions = wavesurfer.registerPlugin(Regions.create());
let color = randomColor();
let dragSelection = wsRegions.enableDragSelection({
	color: color,
	content: `New label`
});

wsRegions.on('region-created', (region: any) => {
	dragSelection();
	color = randomColor();
	dragSelection = wsRegions.enableDragSelection({
		color: color,
		content: `New label`
	});
});
```
This works but I am wondering if there is a better method to do this, or if changes must be made to the plugin?

**Comment by katspaugh** [comment, marked as answer]
You can call `region.setOptions` to set a random color on creation.

```js
wsRegions.on('region-created', (region) => {
  region.setOptions({
    color: randomColor()
  })
})
